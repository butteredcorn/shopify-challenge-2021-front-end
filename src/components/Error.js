import styled from 'styled-components'

const ErrorContainer = styled.strong`
    color: ${props => props.theme.error.primary};
`
const Error = ({message}) => {
    return (
        <ErrorContainer>
            Error: {message}
        </ErrorContainer>
    );
  };
export default Error;