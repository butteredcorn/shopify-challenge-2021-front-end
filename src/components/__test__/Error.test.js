import { render, screen, cleanup } from '@testing-library/react';
import ThemesProvider from '../../providers/ThemeProvider'
import renderer from 'react-test-renderer'
import Error from '../Error';

afterEach(cleanup);

it('renders Error component with message "Hello World!"', () => {
  render(<ThemesProvider><Error message={"Hello World!"}/></ThemesProvider>);
  const message = screen.getByText(/Hello World!/i);
  expect(message).toBeInTheDocument();
});

it('matches a snapshot', () => {
  const json = renderer.create(<ThemesProvider><Error message={"Hello World!"}/></ThemesProvider>).toJSON();
  expect(json).toMatchSnapshot();
});