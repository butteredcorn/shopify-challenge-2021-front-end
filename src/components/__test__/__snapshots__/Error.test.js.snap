// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<strong
  className="sc-bdvvaa hXxLwo"
>
  Error: 
  Hello World!
</strong>
`;
